### ISSUE TITLE
[release  1.18.114] chunk size affect decoding results

### ISSUE THREAD

**Issue by sadahanu** [awslabs/sockeye#907]
Hi, 

We are using an older version of sockeye to do translation En->JA. (1.18.114)
However, we find out that the chunk size affects the translation results, especially for sentences not ending punctuations. 
If we set chunk_size = 1 (the first sentence (no period) is not translated correctly)
I see that the item is out for delivery -> 商品が配達中であることがわかり
I see that the item is out for delivery. -> 商品が配達中であることがわかりました。

If we set chunk_size = 1000 (now this is correct)
I see that the item is out for delivery -> 商品が配達中であることがわかりました
I see that the item is out for delivery. -> 商品が配達中であることがわかりました。

Appreciate any inputs.

**Comment by tdomhan** [comment]
Do use using (`--batch-size`)?

One thing you could try is to manually setting the maximum output length (`--max-output-length`) to a large value, to see whether this has an impact on the translation.

**Comment by sadahanu** [comment]
Thanks for the comments.

For the first comment,  I tried both batch_size = 1 and batch_size=64,  but both give same results as I pasted above (different results with chunk_size). 
As my input is very short (the testset only has four sentences), it is a bit hard for me to understand why chunk_size would play such a big role. 
For the second comment, the max-output-length was set to be None. It seems that when it is none it is set by ```int(np.ceil(factor * input_length))```  - that can be a factor - let me experiment this one out. Thanks a lot!
